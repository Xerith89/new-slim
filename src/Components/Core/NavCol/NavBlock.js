import React, { Component } from 'react'
import {SlideDown} from 'react-slidedown'
import 'react-slidedown/lib/slidedown.css'
import NavItem from './NavItem';
import PropTypes from 'prop-types'

export default class NavBlock extends Component {

    constructor(props) {
        super(props)

        this.state = {childrenVisible: true};
    }
    
    handleClick = () => {
        this.setState({
            childrenVisible: !this.state.childrenVisible
        });
    }

    render() {
        return (
            <div>
                <button className="btn btn-block border" data-testid="navButton" style={{backgroundColor: "#dfe4e6", borderColor: "#868786"}} onClick={this.handleClick}><strong>{this.props.title}</strong></button>
                <SlideDown className={'my-dropdown-slidedown'}>
                    {this.state.childrenVisible ? <NavItem links={this.props.links}/> : null}
                </SlideDown>
            </div>
        )
    }
}

NavBlock.propTypes = {
    title: PropTypes.string,
    links: PropTypes.array
};
