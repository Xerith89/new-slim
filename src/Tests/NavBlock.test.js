import React from 'react';
import NavBlock from '../Components/Core/NavCol/NavBlock';
import {render, cleanup, fireEvent} from '@testing-library/react';
import { faPencilAlt } from '@fortawesome/free-solid-svg-icons';

afterEach(cleanup);

const underwritingLinks = {
    links: [
        "New Quote",
        "Renewals",
        "Another Link",
    ]
}

test('NavBlock component renders to the DOM', () => {
    const {getByText} = render(<NavBlock icon={faPencilAlt}title="Underwriting" links={underwritingLinks.links} />);
    expect(getByText('Underwriting')).toMatchSnapshot();
});

