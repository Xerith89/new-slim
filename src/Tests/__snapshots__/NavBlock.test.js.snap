// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBlock component renders to the DOM 1`] = `
<strong>
   
  Underwriting
</strong>
`;
